Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemScriptingHandler.cpp

```diff
@@ -72,6 +74,9 @@ namespace AzToolsFramework::Prefab
             entities, commonRoot, &topLevelEntities);
 
         auto containerEntity = AZStd::make_unique<AZ::Entity>();
+        containerEntity->CreateComponent<Components::TransformComponent>();
+        containerEntity->CreateComponent<Components::EditorLockComponent>();
+        containerEntity->CreateComponent<Components::EditorVisibilityComponent>();
```

Comment:
For an editor entity, a [lot more components](https://github.com/o3de/o3de/blob/df9b4d4a2fb197c551b1db66f5befd716797c614/Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EditorEntityContextComponent.cpp#L129-L138) gets added by the editor entity context by default. I wonder if they should be added here too to avoid running into some missing editor component in the future.

Reply:
Likely. When verifying the current functions creating editor entities, these were the components that were absolutely necessary - the rest got added in the Editor with no additional errors. It would be useful to set a standard way of creating an entity programmatically though so that these kind of issues are circumvented.